kubespawner/objects.py

```diff
@@ -566,3 +596,107 @@ def make_ingress(
     )
 
     return endpoint, service, ingress
+
+
+def make_secret(
+    name,
+    username,
+    ssl_directory,
+    labels=None,
+    annotations=None,
+):
+    """
+    Make a k8s secret specification for mount ssl credentials for running a user notebook.
```

Comment:
If my understanding is correct, this function does not generate key/crt files while doing this, but will instead assume they exist by now in the `ssl_directory` folder.

Reply:
Correct it does not generate the key/crt.  It uses the existing key/crt created by JupyterHub to create a kubernetes secret.  If you have better wording for that doc I'm all for it.

Reply:
```suggestion
    Make a k8s secret specification using pre-existing ssl credentials for a given user.
```